arch/arm/core/cortex_m/mpu/arm_mpu.c

```diff
@@ -415,14 +392,15 @@ int arm_core_mpu_buffer_validate(void *addr, size_t size, int write)
  * This function provides the default configuration mechanism for the Memory
  * Protection Unit (MPU).
  */
-static void _arm_mpu_config(void)
+static int arm_mpu_init(struct device *arg)
 {
 	u32_t r_index;
 
-	/* ARM MPU supports up to 16 Regions */
```

Comment:
This was probably wrong info: AFAIK, Cortex-M4 PU supports up to 8 regions.

Reply:
i saw this in a document somewhere.  You can have more than 8 regions, it's not capped to 8 but it is implementation specific.

Reply:
@agross-linaro I think you're right, ARMv7-m allows more regions. Didn't find the info that limits this. MPU_TYPE.DREGION is 8-b long, so it needs to be at most 256.

However, Cortex-M3,4 implement 8, so the comment was a bit confusing, for most of developers that work on M4 and M4